ironfish/src/wallet/recentFeeCache.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Assert } from '../assert'
+import { Blockchain } from '../blockchain'
+import { createRootLogger, Logger } from '../logger'
+import { PriorityQueue } from '../memPool'
+import { MetricsMonitor } from '../metrics'
+import { Block, Transaction } from '../primitives'
+import { Queue } from './queue'
+
+export class RecentFeeCache {
+  private queue: Queue<Transaction>
```

Comment:
Do we need anything from the transaction in the queue besides the fee rate? I think this could be a queue of numbers

Reply:
On second thought we may need some transaction data (e.g., block hash) to handle removing transactions when a forked block is disconnected from the chain

Reply:
sounds good!